File: LeetCode/0066/golang/lc0066_test.go
package lc0066

import (
	"slices"
	"testing"
)

func Test_plusOne(t *testing.T) {
	testcases := []struct {
		digits   []int
		expected []int
	}{
		{[]int{1, 2, 3}, []int{1, 2, 4}},
		{[]int{4, 3, 2, 1}, []int{4, 3, 2, 2}},
		{[]int{9}, []int{1, 0}},
		{[]int{0}, []int{1}},
		{[]int{1, 9}, []int{2, 0}},
		{[]int{8, 9}, []int{9, 0}},
		{[]int{9, 9}, []int{1, 0, 0}},
	}

	for i, tc := range testcases {
		actual := plusOne(tc.digits)
		if !slices.Equal(actual, tc.expected) {
			t.Errorf("Testcase %02d (%v) failed: want %v, got %v",
				i+1, tc.digits, tc.expected, actual)
		}
	}
}

File: LeetCode/0066/python/lc0066_test.py
import unittest

from lc0066 import Solution

class Test_plusOne(unittest.TestCase):
    def test(self):
        s = Solution()
        self.assertEqual(s.plusOne([1, 2, 3]), [1, 2, 4])
        self.assertEqual(s.plusOne([4, 3, 2, 1]), [4, 3, 2, 2])
        self.assertEqual(s.plusOne([9]), [1, 0])
        self.assertEqual(s.plusOne([0]), [1])
        self.assertEqual(s.plusOne([1, 9]), [2, 0])
        self.assertEqual(s.plusOne([8, 9]), [9, 0])
        self.assertEqual(s.plusOne([9, 9]), [1, 0, 0])

if __name__ == "__main__":
    unittest.main()
